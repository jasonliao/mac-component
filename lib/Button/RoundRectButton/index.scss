.round-rect-button {
  box-sizing: border-box;
  color: #222831;
  border: 1px solid #a5a5a5;
  background: #ececec;
  border-radius: 4px;
  outline: none;
  padding: 1px 14px;
}

.round-rect-button:active {
  background: #d4d4d4;
}

.round-rect-button.round-rect-button__disabled,
.round-rect-button.round-rect-button__disabled:active {
  background: #ececec;
  border: 1px solid #c8c8c8;
  color: #c8c8c8;
}

.round-rect-button__regular {
  font-size: 13px;
}

.round-rect-button__small {
  font-size: 11px;
}

.round-rect-button__mini {
  font-size: 9px;
}