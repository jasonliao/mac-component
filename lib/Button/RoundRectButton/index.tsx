import * as React from 'react'
import classNames from 'classnames'

import Button, { ButtonProps } from '../BaseComponents/Button'

import './index.scss'

export interface RoundRectButtonProps extends ButtonProps {}

export default class RoundRectButton extends Button<RoundRectButtonProps> {
  render() {
    const { size, disabled, style, className, children } = this.props

    const cls = classNames(
      className,
      'round-rect-button',
      `round-rect-button__${size}`,
      {
        'round-rect-button__disabled': disabled,
      }
    )

    return (
      <button
        className={cls}
        style={style}
        onClick={this.handleClick}
      >
        { children }
      </button>
    )
  }
}