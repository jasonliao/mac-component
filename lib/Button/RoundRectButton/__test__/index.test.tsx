import * as React from 'react'
import { shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'

import RoundRectButton, { RoundRectButtonProps } from '../index'

describe('RoundRectButton Component', () => {
  const content = 'Textured Round Button'
  it('[UI] render text', () => {
    const wrapper = shallow(<RoundRectButton>{ content }</RoundRectButton> )

    const text = wrapper.find('button').text()
    expect(text).toBe(content)
  })

  it('[UI] render with mini size', () => {
    const props: RoundRectButtonProps = { size: 'mini', }
    const wrapper = shallow(<RoundRectButton { ...props } />)

    const hasClass = wrapper.find('.round-rect-button').hasClass(`round-rect-button__${props.size}`)
    expect(hasClass).toBe(true)
  })

  it('[UI] render disabled button', () => {
    const props: RoundRectButtonProps = { disabled: true }
    const wrapper = shallow(<RoundRectButton { ...props } />)

    const hasClass = wrapper.find('.round-rect-button').hasClass(`round-rect-button__disabled`)
    expect(hasClass).toBe(true)
  })

  it('[BEHAVIOR] button clicked', () => {
    const props: RoundRectButtonProps = { onClick: jest.fn() }
    const wrapper = shallow(<RoundRectButton { ...props } />)

    wrapper.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })

  it('[BEHAVIOR] disabled button clicked', () => {
    const props: RoundRectButtonProps = { disabled: true, onClick: jest.fn() }
    const wrapper = shallow(<RoundRectButton { ...props } />)

    wrapper.simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(0)
  })

  it('[SNAPSHOT] match the snapshot', () => {
    const props: RoundRectButtonProps = {
      size: 'small',
      style: { width: '100px' },
      className: 'custom_classname'
    }

    const tree = renderer.create(<RoundRectButton { ...props } >{ content }</RoundRectButton>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})