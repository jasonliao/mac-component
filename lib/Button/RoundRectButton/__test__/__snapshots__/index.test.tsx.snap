// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundRectButton Component [SNAPSHOT] match the snapshot 1`] = `
<button
  className="custom_classname round-rect-button round-rect-button__small"
  onClick={[Function]}
  style={
    Object {
      "width": "100px",
    }
  }
>
  Textured Round Button
</button>
`;
